api/service.go

```diff
@@ -45,36 +50,257 @@ func (s *service) RegisterSchema(ctx context.Context, req *pb.RegisterSchemaRequ
 	return &pb.RegisterSchemaReply{}, nil
 }
 
+func (s *service) Start(ctx context.Context, req *pb.StartRequest) (*pb.StartReply, error) {
+	store, err := s.getStore(req.GetStoreID())
+	if err != nil {
+		return nil, err
+	}
+	if err := store.Start(); err != nil {
+		return nil, err
+	}
+	return &pb.StartReply{}, nil
+}
+
+func (s *service) StartFromAddress(ctx context.Context, req *pb.StartFromAddressRequest) (*pb.StartFromAddressReply, error) {
+	var err error
+	var store *es.Store
+	var addr ma.Multiaddr
+	var readKey, followKey *symmetric.Key
+	if store, err = s.getStore(req.GetStoreID()); err != nil {
+		return nil, err
+	}
+	if addr, err = ma.NewMultiaddr(req.GetAddress()); err != nil {
+		return nil, err
+	}
+	if readKey, err = symmetric.NewKey(req.GetReadKey()); err != nil {
+		return nil, err
+	}
+	if followKey, err = symmetric.NewKey(req.GetFollowKey()); err != nil {
+		return nil, err
+	}
+	if err = store.StartFromAddr(addr, followKey, readKey); err != nil {
+		return nil, err
+	}
+	return &pb.StartFromAddressReply{}, nil
+}
+
 // ModelCreate adds a new instance of a model to a store.
 func (s *service) ModelCreate(ctx context.Context, req *pb.ModelCreateRequest) (*pb.ModelCreateReply, error) {
 	log.Debugf("received model create request for model %s", req.ModelName)
+	model, err := s.getModel(req.StoreID, req.ModelName)
+	if err != nil {
+		return nil, status.Error(codes.NotFound, "model not found")
+	}
+	return s.processCreateRequest(req, model.Create)
+}
 
-	store, err := s.getStore(req.StoreID)
+func (s *service) ModelSave(ctx context.Context, req *pb.ModelSaveRequest) (*pb.ModelSaveReply, error) {
+	model, err := s.getModel(req.StoreID, req.ModelName)
 	if err != nil {
-		return nil, err
+		return nil, status.Error(codes.NotFound, "model not found")
```

Comment:
This `err` should already be correctly wrapped in `getModel`. It could be "store" not found as well.

Reply:
Nice catch, thx